addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -1684,9 +1684,9 @@ def test_accrual_period_start_past_start_date(self):
             with Form(self.env['hr.leave.allocation']) as f:
                 f.allocation_type = "accrual"
                 f.accrual_plan_id = accrual_plan
+                f.date_from = '2024-01-01'
                 f.employee_ids.add(self.employee_emp)
                 f.holiday_status_id = self.leave_type
-                f.date_from = '2024-01-01'
```

Comment:
This test was failing and when I checked why it is failing I found out that when adding the employee in this test it actually computes the number of days while it shouldn't because of this condition https://github.com/odoo/odoo/blob/17.0/addons/hr_holidays/models/hr_leave_allocation.py#L862:L863
but seems like the form has a FakeDate and it gets passed which cause that the number of days gets computed twice once when adding the employee and another and other when adding the date.

By changing this line we are actually covering both my fix and the use case this test was checking before